### PR TITLE
Remove JVM version dependent GC setting handling

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

We require 11+, so we no longer need to handle <= 1.8 style GC flags

## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: app/src/bin/crate.in.sh [hunk 1]
```diff
@@ -80,26 +80,7 @@ EOF
   else
       JAVA=java
   fi
-  JAVA_VERSION=`$JAVA -version 2>&1 | awk '/version/ {gsub(/"/, "", $3); split($3, parts, ".")} END {print (parts[1] == 1 ? parts[2] : parts[1])}'`
-
-  if [ "x$JAVA_VERSION" = "x" ]; then
-    echo 'Java is missing or $JAVA_HOME is not set correctly. Install a JRE or setup $JAVA_HOME'
-    exit 1
-  fi
```
Comment (seut) [inline app/src/bin/crate.in.sh]: how is this related? shouldn't we still complain when no java is found?

Comment (mfussenegger) [reply app/src/bin/crate.in.sh]: You're right that it is a bit unrelated.

Now we get

```
❮ PATH=/usr/bin/ JAVA_HOME=/tmp bin/crate
bin/crate: line 129: exec: java: not found
```

Before it was:

```
❮ PATH=/usr/bin/ JAVA_HOME=/tmp bin/crate       
Java is missing or $JAVA_HOME is not set correctly. Install a JRE or setup $JAVA_HOME
```

I removed it because it was using the `$JAVA_VERSION` to determine if it should be logged and we didn't have the same logic in the `bat` script.

Comment (seut) [reply app/src/bin/crate.in.sh]: Not sure if this is worth re-adding it, just wanted to mention that we somehow change a behaviour slightly. I have no strong opinion here as the `java not found` message may be enough to understand the issue.
